### ISSUE TITLE
Add support for filters and sorting

### ISSUE THREAD

**Issue by arvind0598** [arvind0598/better-first-issues#8]
The GitHub Search API allows for passing multiple parameters, including filters and sorting related stuff. We should build the frontend components to get that data from the user and enhance the API in the backend to parse that data too.

**Comment by G-ala-Hub** [comment]
Hi @arvind0598 could you please assign this to me thanks

**Comment by G-ala-Hub** [comment]
@arvind0598 I have started working on the frontend side. is this what you had in mind ?
<img width="745" alt="Screen Shot 2023-07-16 at 5 02 17 pm" src="https://github.com/arvind0598/better-first-issues/assets/91773328/8a703942-cf57-4c1e-9ecf-61f4fe7c64bf">

**Comment by arvind0598** [comment]
Not exactly. Your input supports "tags" essentially, but the API supports a lot of different parameters. You'd have to do something similar to what you'd find on an ecommerce site, having separate filters for "repository size", "repository language" etc.

You can check these [docs](https://docs.github.com/en/search-github/searching-on-github/searching-for-repositories) for more info.
